File: euler/p19.ml

let count () =
  let n = [| 31; 28; 31; 30; 31; 30; 31; 31; 30; 31; 30; 31 |] in
  let compte = ref 0 in
  let day_of_the_week = ref 1 in
  for year = 1901 to 2000 do
    for month = 1 to 12 do
      let nb_days = (if year mod 4 = 0 && month = 2 then 29 else n.(month - 1)) in
      for day = 1 to nb_days do
        compte := !compte + (if !day_of_the_week = 6 && day = 1 then 1 else 0);
        day_of_the_week := (!day_of_the_week + 1) mod 7;
      done;
    done;
  done;
  !compte

File: euler/p19.c
#include <stdio.h>

int main() {
  int n_days[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

  int compte = 0;

  int day_of_the_week = 1;// modulo 7

  for (int year = 1901; year <= 2000; year++) {
    for (int month = 1; month <= 12; month++) {
      int nb_days = n_days[month - 1];

      if (year % 4 == 0 && month == 2) nb_days = 29;

      for (int day = 1; day <= nb_days; day++) {
        if (day == 1 && day_of_the_week == 6) compte++;
        day_of_the_week = (day_of_the_week + 1) % 7;
      }
    }
  }

  printf("Compte : %d\n", compte);
}
